core/driver/sources/src/main/scala/org/virtuslab/ideprobe/dependencies/ResourceProvider.scala

```diff
@@ -73,7 +74,11 @@ object ResourceProvider {
     }
 
     private def cached(uri: URI): Path = {
-      directory.resolve(Hash.md5(uri.toString))
+      val resultPath = directory.resolve(Hash.md5(uri.toString))
+      if (uri.toString.endsWith(".dmg"))
```

Comment:
you could make it more generic - to just keep extensions on these files. It would make checking this cache manually easier as well.